### PR TITLE
Provide an app entrypoint to DWDS

### PR DESCRIPTION
**Pull request by elliette**
Work towards https://github.com/flutter/devtools/issues/5350

Requires the following changes to Flutter: https://github.com/flutter/flutter/compare/master...elliette:dwdsEntrypoint

### CODE REVIEW

File: dwds/lib/src/debugging/inspector.dart [hunk 1]
```diff
@@ -136,6 +136,7 @@ class AppInspector implements AppInspectorInterface {
     String root,
     Debugger debugger,
     ExecutionContext executionContext,
+    LoadStrategy loadStrategy,
```
Comment (annagrin) [inline dwds/lib/src/debugging/inspector.dart:139]: Btw we have the load strategy available as a `globalLoadStrategy` (set on `dwds.start` so we don't need to pass it around. Can we switch to using the global in this PR as well? Alternatively, we can move to passing load strategy everywhere (not sure the work involved is justifiable though).

**Review by annagrin — COMMENTED** [review]
Thanks Elliott, added a comment.

File: dwds/test/fixtures/context.dart [hunk 2]
```diff
@@ -90,6 +90,13 @@ class TestContext {
         ),
       );
 
+  /// The package URI of the Dart entry file, e.g,
+  /// "org-dartlang-app:example/hello_world/main.dart":
+  Uri get _dartEntryFilePackageUri => Uri.parse('org-dartlang-app:///${p.join(
```
Comment (annagrin) [inline dwds/test/fixtures/context.dart:95]: Can this getter be moved to the TestProject?

**Review by annagrin — APPROVED** [review]
LGTM with a small comment

File: dwds/test/fixtures/context.dart (hunk 2 above)
Comment (elliette) [reply dwds/test/fixtures/context.dart]: Done!

File: dwds/lib/src/debugging/inspector.dart (hunk 1 above)
Comment (elliette) [reply dwds/lib/src/debugging/inspector.dart]: Done, thanks!

**Comment by elliette** [comment]
Adding @christopherfujino as well

This is a breaking change and will require the following change in `flutter_tools` when updating to the next version of DWDS: https://github.com/flutter/flutter/compare/master...elliette:dwdsEntrypoint

**Comment by elliette** [comment]
> Adding @christopherfujino as well
> 
> This is a breaking change and will require the following change in `flutter_tools` when updating to the next version of DWDS: [flutter/flutter@master...elliette:dwdsEntrypoint](https://github.com/flutter/flutter/compare/master...elliette:dwdsEntrypoint)

Chris is OOO, will hold off until next week to release new version.

**Pull request merged**